### PR TITLE
setting things up for pypi

### PR DESCRIPTION
**Pull request by hstojic**
2 things I would need some help with:
- pyproject.toml - [build-system] currently points to poetry, is that fine for building a package for pip?
- I'm not convinced we need all the libraries listed in `install_requires` in `setup.py` - @johnamcleod  you were taking care of dependencies before, can you give a hand here please?

I have set up a workflow for psuhing things to PyPi automatically, not sure how to test it though (hm, perhaps I could modfiy it to use test PyPi...)
I will first push things to test PyPi, to verify things work as intended

### CODE REVIEW

**Review by johnamcleod — COMMENTED** [review]
> * pyproject.toml - [build-system] currently points to poetry, is that fine for building a package for pip?

Yes, `pip` will ignore the `pyproject.toml` file. 

> * I'm not convinced we need all the libraries listed in `install_requires` in `setup.py` - @johnamcleod  you were taking care of dependencies before, can you give a hand here please?

True. I will look at that today and trim the list of dependencies.

File: .github/workflows/upload-pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+on:
+  push:
+    tags: v[0-9]+.[0-9]+.[0-9]+*
```
[the remark this answers is not in the document]
Comment (johnamcleod) [reply .github/workflows/upload-pypi.yaml:17]: GitHub is following this git change: https://sfconservancy.org/news/2020/jun/23/gitbranchname/. A new GitHub repository has a `main` branch as the initial default branch.

**Comment by hstojic** [comment]
@johnamcleod all good now?

**Review by johnamcleod — APPROVED** [review]
Looks good to me!

**Pull request merged**